src/main/scala/io/iohk/ethereum/mpt/MptVisitors/MptConstructionVisitor.scala

```diff
@@ -0,0 +1,54 @@
+package io.iohk.ethereum.mpt.MptVisitors
+
+import io.iohk.ethereum.db.storage.MptStorage
+import io.iohk.ethereum.db.storage.NodeStorage.NodeHash
+import io.iohk.ethereum.mpt.{BranchNode, ExtensionNode, HashNode, LeafNode, MptNode, NullNode}
+
+class MptConstructionVisitor(source: MptStorage) extends MptVisitor[MptNode] {
+
+  def visitLeaf(leaf: LeafNode): MptNode = {
+    leaf
+  }
+
+  def visitHash(hashNode: HashNode): HashNodeResult[MptNode] = {
+    ResolveResult(source.get(hashNode.hash))
+  }
+
+  override def visitNull(): MptNode = {
+    NullNode
+  }
+
+  override def visitExtension(extension: ExtensionNode): ExtensionVisitor[MptNode] = new MptExtensionVistor(extension, source)
+
+  override def visitBranch(value: BranchNode): BranchVisitor[MptNode] = new MptBranchVistor(value, source)
+}
+
+class MptBranchVistor(branchNode: BranchNode, source: MptStorage) extends BranchVisitor[MptNode] {
```

Comment:
I know that you just extracted it from another file but maybe you could fix typo for `MptBranchVistor` and `MptExtensionVistor`, because both are missing `i`